functions/migrate.go

```diff
@@ -35,58 +35,71 @@ func Migrate() {
 		fmt.Println("error reading network data ", err.Error())
 		return
 	}
-	logger.Log(0, "stoping daemon")
+	if len(networks) == 0 { // nothing to migrate
+		return
+	}
+	logger.Log(0, "stopping daemon")
 	if err := daemon.Stop(); err != nil {
 		logger.Log(0, "failed to stop daemon", err.Error())
 	}
+
 	logger.Log(0, "networks to be migrated", strings.Join(networks, " "))
+	var legacyNodes = []models.LegacyNode{}
+	var servers = map[string]struct{}{}
+	var newHost models.Host
 	for _, network := range networks {
 		logger.Log(0, "migrating", network)
 		cfg, err := config.ReadConfig(network)
 		if err != nil {
 			logger.Log(0, "failed to read config for network", network, "skipping ...")
 			continue
 		}
-		pass, err := os.ReadFile(config.GetNetclientPath() + "config/secret-" + network)
+		oldIface := cfg.Node.Interface
+		secretPath := config.GetNetclientPath() + "config/secret-" + network
+		if ncutils.IsWindows() {
+			secretPath = config.GetNetclientPath() + "secret-" + network
+		}
+		pass, err := os.ReadFile(secretPath)
 		if err != nil {
 			logger.Log(0, "could not read secrets file", err.Error())
 			continue
 		}
+		cfg.Node.Password = string(pass)
+		legacyNodes = append(legacyNodes, cfg.Node)
 		node, netclient := config.ConvertOldNode(&cfg.Node)
 		node.Server = strings.Replace(cfg.Server.Server, "broker.", "", 1)
-		serverHost, serverNode := config.Convert(netclient, node)
+		servers[cfg.Server.API] = struct{}{}
+		newHost, _ = config.Convert(netclient, node)
+		newHost.PublicKey = netclient.PrivateKey.PublicKey()
 		ip, err := getInterfaces()
 		if err != nil {
 			logger.Log(0, "failed to retrieve local interfaces", err.Error())
 		} else {
 			// just in case getInterfaces() returned nil, nil
 			if ip != nil {
-				serverHost.Interfaces = *ip
+				newHost.Interfaces = *ip
 			}
 		}
 		defaultInterface, err := getDefaultInterface()
 		if err != nil {
 			logger.Log(0, "default gateway not found", err.Error())
 		} else {
-			serverHost.DefaultInterface = defaultInterface
+			newHost.DefaultInterface = defaultInterface
 		}
-		serverNode.ID = uuid.Nil
+		wireguard.DeleteOldInterface(oldIface)
+	}
+	var serversToWrite = []models.ServerConfig{}
+	var hostToWrite *models.Host
+	for k := range servers {
+		server := k
+		logger.Log(0, "migrating for server", server)
 		migrationData := models.MigrationData{
-			JoinData: models.JoinData{
-				Node: serverNode,
-				Host: serverHost,
-			},
-			LegacyNodeID: cfg.Node.ID,
-			Password:     string(pass),
-		}
-		//call migrate node
-		if config.Netclient().Debug {
-			fmt.Println("calling migration endpoint for node", cfg.Node.ID)
-			pretty.Println(migrationData)
+			LegacyNodes: legacyNodes,
```

Comment:
if the old client was connected to multiple servers all old nodes (not just the ones from the server in question) will be sent.  Netmaker will just ignore the non-existent (from its point of view) node.